wgpu-core/src/command/mod.rs

```diff
@@ -270,3 +270,19 @@ where
         count_words += size_to_write_words;
     }
 }
+
+#[derive(Debug)]
+struct StateChange<T> {
+    last_state: Option<T>,
+}
+
+impl<T: Copy + PartialEq> StateChange<T> {
+    fn new() -> Self {
+        Self { last_state: None }
+    }
+    fn already_set(&mut self, new_state: T) -> bool {
```

Comment:
this method is mutating, but it's named as if it's an immutable check, which is confusing.
Maybe we can call it just `set`?